process/smartContract/hooks/blockChainHook.go

```diff
@@ -440,6 +487,24 @@ func (bh *BlockChainHookImpl) ProcessBuiltInFunction(input *vmcommon.ContractCal
 	return vmOutput, nil
 }
 
+func (bh *BlockChainHookImpl) checkMaxCounters(input *vmcommon.ContractCallInput) error {
+	if bh.enableEpochsHandler.IsMaxBlockchainHookCountersFlagEnabled() {
+		bh.crtNumberOfBuiltInFunctionCalls++
+		if bh.crtNumberOfBuiltInFunctionCalls > bh.maxBuiltInCallsPerTx {
+			return process.OutOfAPICalls
```

Comment:
suggestion for all OutOfAPICalls returns (for better user experience and easier contract debugging)
```GO
fmt.Errorf("%w, crtNumberOfBuiltInFunctionCalls: %d, maxBuiltInCallsPerTx: %d", process.OutOfAPICalls, 
bh.crtNumberOfBuiltInFunctionCalls, bh.maxBuiltInCallsPerTx)
```
I know this requirement will create errors that will be hard to further change (since the nodes will apply the consensus upon them) but I believe it will greatly help the developers since we have 3 counters that can overflow.

Reply:
done